src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+            throw new DataProcessingException("Couldn't get car by id "
+                    + id + ". ", e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id AS car_id, model, m.id AS manufacturer_id,"
+                + " m.name AS manufacturer_name, m.country AS manufacturer_country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "AND c.is_deleted = FALSE;";
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carList.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of all cars from DB.", e);
+        }
+        for (Car car: carList) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return carList;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, car.getModel());
+            statement.setObject(2, car.getManufacturer().getId());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in DB.", e);
+        }
+        removeDriversFromCar(car.getId());
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        boolean deleted;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            deleted = statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+        removeDriversFromCar(id);
+        return deleted;
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getCarsByDriverQuery = "SELECT id, model, manufacturer_id "
+                + "FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "WHERE cd.driver_id = ? AND is_deleted = FALSE;";
```

Comment:
I think there will be exception on line 150 ? How do you set manufacturer_name if you never query it ?

Reply:
Fixed, i'm so impressed that you find it :fire: